### PR TITLE
[core] Improve python code

### PR DESCRIPTION
**Pull request by KlicOgogo**

### CODE REVIEW

File: core/common.py [hunk 1]
```diff
@@ -7,6 +10,8 @@ def date_is_correct(date):
     for date_item in date_items:
         if not date_item.isdigit():
             return False
+    if len(date_items[1]) != 2 or len(date_items[2]) != 2:
```
Comment (Drapegnik) [inline core/common.py]:
```suggestion
    if list(map(len, date_items[1:2])) != [2, 2]
```

File: core/common.py [hunk 2]
```diff
@@ -27,3 +32,11 @@ def date_is_correct(date):
             return day > 0 and day < 29
         else:
             return day > 0 and day < 30
+
+
+def today():
+    cur_date = datetime.now()
+    cur_year = str(cur_date.year)
+    cur_month = '0' * (cur_date.month < 10) + str(cur_date.month)
```
Comment (Drapegnik) [inline core/common.py]:
```suggestion
    cur_month = f'{cur_date.month}'.rjust(2, '0')
```

File: core/get_ratings.py [hunk 3]
```diff
@@ -1,148 +1,141 @@
...
 
 
 def get_norm_coeff(old_rating, sets_played):
-    if sets_played < 100:
-        return 40
-    elif old_rating >= 2400:
-        return 10
-    else:
-        return 20
+    return 40 if sets_played < 100 else 10 if old_rating >= 2400 else 20
```
Comment (Drapegnik) [inline core/get_ratings.py]: maybe extract `100` and `2400` to constants?

File: core/get_ratings.py [hunk 4]
```diff
@@ -1,148 +1,141 @@
...
+
     for i in range(n_games):
         new_ratings = update_ratings_with_game(ratings_log[-1][1], proper_games.loc[i], set_counts)
         new_ratings = new_ratings.sort_values(['Rating'], ascending=False)
         ratings_log.append((proper_games.loc[i]['Date'], new_ratings.copy(deep=True)))
     return ratings_log[-1][1], ratings_log, set_counts
 
 
+def get_previous_rating(name, ratings_log):
+    for i in range(len(ratings_log)):
```
Comment (Drapegnik) [inline core/get_ratings.py]:
```suggestion
    for (i, (cur_day, cur_ratings)) in enumerate(ratings_log):
```

File: core/get_ratings.py [hunk 5]
```diff
@@ -1,148 +1,141 @@
...
+        cur_day, cur_ratings = ratings_log[i]
+        if cur_day == ratings_log[-1][0]:
```
Comment (Drapegnik) [inline core/get_ratings.py]: what's the point?
`cur_day == last_day`?

File: core/get_ratings.py [hunk 6]
```diff
@@ -1,148 +1,141 @@
...
+            if i != 0:
+                prev_day, prev_ratings = ratings_log[i-1]
+                return int(np.round(prev_ratings.loc[name]['Rating']))
+            else:
+                return int(np.round(cur_ratings.loc[name]['Rating']))
```
Comment (Drapegnik) [inline core/get_ratings.py]: requires refactor

File: core/get_ratings.py [hunk 7]
```diff
@@ -1,148 +1,141 @@
...
-                    prev_day, prev_ratings = ratings_log[i-1]
-                    dict_to_save['prev_rating'] = int(np.round(prev_ratings.loc[name]['Rating']))
-                else:
-                    dict_to_save['prev_rating'] = int(np.round(cur_ratings.loc[name]['Rating']))
-                break
-        json_list.append(dict_to_save)
-    json_filename = os.path.join(repo_root_dir, 'leaderboard-ui/src/rating.json')
+                        'sets': int(set_counts[name]),
+                        'prev_rating': get_previous_rating(name, ratings_log)}
+        json_data.append(dict_to_save)
```
Comment (Drapegnik) [inline core/get_ratings.py]: could you replace it with `map`?

File: core/get_ratings.py [hunk 8]
```diff
@@ -1,148 +1,141 @@
...
 
 
 def save_ratings_history_to_json(ratings_log):
-    json_list = []
+    json_data = []
     for (date, ratings) in ratings_log:
         ratings_dict = {name: int(np.round(ratings.loc[name]['Rating'])) for name in sorted(ratings.index)}
-        json_list.append({'date': date, 'ratings': ratings_dict})
-    json_filename = os.path.join(repo_root_dir, 'leaderboard-ui/src/ratings_history.json')
+        json_data.append({'date': date, 'ratings': ratings_dict})
```
Comment (Drapegnik) [inline core/get_ratings.py]: same as above

File: core/logger.py [hunk 9]
```diff
@@ -1,48 +1,43 @@
...
     known_players = set(games['Winner'].tolist() + games['Looser'].tolist())
     if player not in known_players:
         print('player {} has no mathups at database, Do you want to add him? ([y]/n)'.format(player))
         confirm = sys.stdin.readline().strip()
-        if confirm == 'n':
-            print('OK')
-            return False
-        elif confirm != 'y' and confirm != '':
-            print('Response is incorrect')
+        if confirm == 'n' or confirm not in ['y', '']:
```
Comment (Drapegnik) [inline core/logger.py:30]: `confirm not in ['y', '']` and `confirm == 'n'` are equal, just keep one)

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: core/common.py (hunk 1 above)
Comment (KlicOgogo) [reply core/common.py]: There should be [1:3], but I will use this

File: core/get_ratings.py (hunk 3 above)
Comment (KlicOgogo) [reply core/get_ratings.py]: OK

File: core/get_ratings.py (hunk 5 above)
Comment (KlicOgogo) [reply core/get_ratings.py]: Yes, but I need first occurrence of last_day to find previous match day

File: core/get_ratings.py (hunk 7 above)
Comment (KlicOgogo) [reply core/get_ratings.py]: Did not understand. Do you mean `[value for _ in _]` syntax?

Comment (KlicOgogo) [reply core/get_ratings.py]: Wouldn't it be too complicated?

File: core/logger.py (hunk 9 above)
Comment (KlicOgogo) [reply core/logger.py:30]: What about incorrect responses?

File: core/get_ratings.py (hunk 7 above)
Comment (Drapegnik) [reply core/get_ratings.py]:
```python
arr = []
for ...
	arr.push()
```

is equal to purest version:

```python
arr = list(map())
```

File: core/logger.py (hunk 9 above)
Comment (Drapegnik) [reply core/logger.py:30]: everything except `y` and `''` is `n`

Comment (KlicOgogo) [reply core/logger.py:30]: User can type everything, not only 'y', 'n' and '', for me such responses are incorrect (I abort them), see previous version of code.

File: core/get_ratings.py (hunk 7 above)
Comment (KlicOgogo) [reply core/get_ratings.py]: So I have to implement dict extractor and use `list(map(my_func, data_to_iterate)`?

Comment (Drapegnik) [reply core/get_ratings.py]:
```python
json_data = list(map(lambda name: {
	'name': name,
	'rating': int(np.round(ratings.loc[name]['Rating'])),
	'sets': int(set_counts[name]),
	'prev_rating': get_previous_rating(name, ratings_log)
}, ratings.index))
```

File: core/get_ratings.py (hunk 5 above)
Comment (Drapegnik) [reply core/get_ratings.py]: Is it possibly to refactor this code in more understandable way?

**Review by Drapegnik — APPROVED** [review]

**Review by Drapegnik — CHANGES_REQUESTED** [review]
.

File: core/get_ratings.py (hunk 5 above)
Comment (KlicOgogo) [reply core/get_ratings.py]: I think, yes

File: core/get_ratings.py [hunk 10]
```diff
@@ -92,31 +92,36 @@ def count_ratings(date):
...
 
 def save_ratings_to_json(ratings, ratings_log, set_counts):
-    json_data = []
-    for name in ratings.index:
-        dict_to_save = {'name': name,
-                        'rating': int(np.round(ratings.loc[name]['Rating'])),
-                        'sets': int(set_counts[name]),
-                        'prev_rating': get_previous_rating(name, ratings_log)}
-        json_data.append(dict_to_save)
+    json_data = list(map(lambda name: {'name': name,
```
Comment (Drapegnik) [inline core/get_ratings.py:106]: do you use some autoformatter?

**Review by Drapegnik — APPROVED** [review]

File: core/get_ratings.py (hunk 10 above)
Comment (KlicOgogo) [reply core/get_ratings.py:106]: No, only python linter package for atom

**Pull request merged**